internal/build/buildconfig/maker.go

```diff
@@ -32,14 +32,25 @@ func NewMaker(helper build.Helper, scheme *runtime.Scheme) Maker {
 	}
 }
 
-func (m *maker) MakeBuildConfig(mod kmmv1beta1.Module, mapping kmmv1beta1.KernelMapping, targetKernel, containerImage string) (*buildv1.BuildConfig, error) {
+func (m *maker) MakeBuildConfig(mod kmmv1beta1.Module, mapping kmmv1beta1.KernelMapping, targetKernel, containerImage string, pushImage bool) (*buildv1.BuildConfig, error) {
 	kmmBuild := m.helper.GetRelevantBuild(mod, mapping)
 
 	buildArgs := m.helper.ApplyBuildArgOverrides(
 		kmmBuild.BuildArgs,
 		kmmv1beta1.BuildArg{Name: "KERNEL_VERSION", Value: targetKernel},
 	)
 
+	buildTarget := buildv1.BuildOutput{
+		To: &v1.ObjectReference{
+			Kind: "DockerImage",
+			Name: containerImage,
+		},
+		PushSecret: mod.Spec.ImageRepoSecret,
+	}
+	if !pushImage {
+		buildTarget = buildv1.BuildOutput{}
+	}
```

Comment:
How about
```suggestion
        buildTarget := buildv1.BuildOutput{}
	if pushImage {
			buildTarget := buildv1.BuildOutput{
		To: &v1.ObjectReference{
			Kind: "DockerImage",
			Name: containerImage,
		},
		PushSecret: mod.Spec.ImageRepoSecret,
	}
	}


```

Reply:
I prefer the original option: in most of the cases (90%) pushImage will be true